include/tvm/runtime/memory.h

```diff
@@ -104,10 +124,22 @@ class SimpleObjAllocator :
       return reinterpret_cast<T*>(data);
     }
 
+    template<typename... Args>
+    static T* NewWithExtraSpace(SimpleObjAllocator*, size_t extra, Args&&... args) {
+      // TODO(wweic): use aligned_alloc in the future
+      void* data = std::malloc(sizeof(T) + extra);
```

Comment:
We care about alignment(otherwise it will leads to mysterious segfaults). 

Please use std::aligned_storage, allocate an array of if. Make sure you at least get alignof(T), and the request alignment is dividable by alignof(T)

Reply:
@tqchen `fields.size()` is unknown until runtime, so `aligned_storage` won't work(it requires the size of memory to be compile time constant) I think. I agree with alignment, that's why I left a note of using `aligned_alloc`. But our c++ support seems to be c++11. `aligned_alloc` is available after c++17. We can have a simple implementation of `aligned_alloc` and switch to it later when we have c++17. What do you think?